common/djangoapps/third_party_auth/saml.py

```diff
@@ -38,6 +39,62 @@ def setting(self, name, default=None):
         except KeyError:
             return self.strategy.setting(name, default, backend=self)
 
+    def get_idp_setting(self, idp, name, default=None):
+        try:
+            return idp.saml_sp_configuration.get_setting(name)
+        except KeyError:
+            return self.setting(name, default)
+
+    def generate_saml_config(self, idp=None):
+        """
+        Override of SAMLAuth.generate_saml_config to use an idp's configured saml_sp_configuration if given.
```

Comment:
This is unfortunate if `generate_saml_config` ever changes but I don't have any better idea

Reply:
we've overwritten several methods here so whenever we upgrade this package, we'll have to make sure to check this stuff.